### ISSUE TITLE
upgrade to 0.3.1 Openshift 

### ISSUE THREAD

**Issue by tonila** [keystonejs/keystone#1118]
After upgrading, app does not have permission to execute. 

I'm getting the following:
    
```
==> app-root/logs/nodejs.log <==
Error: listen EACCES
    at errnoException (net.js:901:11)
    at Server._listen2 (net.js:1020:19)
    at listen (net.js:1061:10)
    at Server.listen (net.js:1135:5)
    at EventEmitter.app.listen (/var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/node_modules/express/lib/application.js:559:24)
    at Object.events.onMount (/var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/lib/core/start.js:128:30)
    at mounted (/var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/lib/core/mount.js:607:29)
    at /var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/lib/updates.js:180:15
    at /var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/async/lib/async.js:157:25
    at Promise.<anonymous> (/var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/lib/updates.js:120:5)
    at Promise.<anonymous> (/var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/node_modules/mongoose/node_modules/mpromise/lib/promise.js:177:8)
    at Promise.EventEmitter.emit (events.js:95:17)
    at Promise.emit (/var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/node_modules/mongoose/node_modules/mpromise/lib/promise.js:84:38)
    at Promise.fulfill (/var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/node_modules/mongoose/node_modules/mpromise/lib/promise.js:97:20)
    at /var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/node_modules/mongoose/lib/query.js:1400:13
    at model.Document.init (/var/lib/openshift/54ee09d0e0b8cdb22b000148/app-root/runtime/repo/node_modules/keystone/node_modules/mongoose/lib/document.js:250:11)
```

I tried setting:

```
if (process.env.OPENSHIFT_NODEJS_IP) {
  keystone.set('host', process.env.OPENSHIFT_NODEJS_IP);
}
if (process.env.OPENSHIFT_NODEJS_PORT) {
  keystone.set('port', process.env.OPENSHIFT_NODEJS_PORT);
}
```

But it does not work either for some reason.

Edit: 
I'm testing with a demo generated with generator-keystone with the default settings and no blogs, galleries or such. 

The only change to make it work with Openshift is setting up the mongo url:
```
if (process.env.OPENSHIFT_MONGODB_DB_URL) {
  keystone.set('mongo', process.env.OPENSHIFT_MONGODB_DB_URL);
}
```
It works well, if I use keystone version 0.2.X but stops working after I change it to 0.3.X.

**Comment by JohnnyEstilles** [comment]
I ran until this issue when I upgraded one of my (non-Keystone) OpenShift Express 3.x  apps to Express 4.x. It had to do with the way I was invoking `app.listen()`. For some reason the OpenShift/Express 4 app required the `host` parameter when calling `app.listen()`.

I had to change it from:
```js
app.list(port, callback);
```
to
```js
app.list(port, host, callback);
```

So, setting Keystone's `host` and `port` options like you mentioned in your comment, plus this change should do the trick.

This will require a change to Keystone ... I'll submit a PR with a suggested fix later today ...
